src/libtriton/format/pe/pe.cpp

```diff
@@ -50,62 +50,69 @@ namespace triton {
 
         this->raw = new(std::nothrow) triton::uint8[this->totalSize];
         if(!this->raw)
-          throw triton::exceptions::PE("PE::open(): Not enough memory.");
+          throw triton::exceptions::Pe("Pe::open(): Not enough memory.");
 
         // Read the file contents
         if (fread(this->raw, 1, this->totalSize, fd) != this->totalSize)
-          throw triton::exceptions::PE("PE::open(): Cannot read the binary file.");
+          throw triton::exceptions::Pe("Pe::open(): Cannot read the binary file.");
 
         // Close the file
         fclose(fd);
       }
 
 
-      bool PE::parse(void) {
+      bool Pe::parse(void) {
         this->header.parse(this->raw, this->totalSize);
         return true;
       }
 
 
-      void PE::initMemoryMapping(void) {
+      void Pe::initMemoryMapping(void) {
         for (auto section : header.getSectionHeaders()) {
           triton::format::MemoryMapping area(this->raw);
 
-          if (this->totalSize < (section.rawAddress + section.rawSize)) {
-            std::cerr << "Warning PE::initMemoryMapping(): Some PE Section Headers of the binary file are corrupted." << std::endl;
+          triton::uint32 rawAddr = section.getRawAddress();
+          triton::uint32 rawSize = section.getRawSize();
+          triton::uint32 virtAddr = section.getVirtualAddress();
+          if (this->totalSize < (rawAddr + rawSize)) {
+            std::cerr << "Warning Pe::initMemoryMapping(): Some PE Section Headers of the binary file are corrupted." << std::endl;
             continue;
           }
 
-          area.setOffset(section.rawAddress);
-          area.setSize(section.rawSize);
-          area.setVirtualAddress(section.virtualAddress);
+          area.setOffset(rawAddr);
+          area.setSize(rawSize);
+          area.setVirtualAddress(virtAddr);
 
           this->memoryMapping.push_back(area);
         }
       }
 
 
-      triton::uint64 PE::getOffsetFromAddress(triton::uint64 vaddr) const {
+      triton::uint64 Pe::getOffsetFromAddress(triton::uint64 vaddr) const {
```

Comment:
Do you have an example where this matters? I thought the alignment only puts constraints on the value of VirtualAddress and RawAddress in the section header, and doesn't affect the calculation that uses these values.